source/grid/grid_tools.cc

```diff
@@ -1573,6 +1573,16 @@ namespace GridTools
       return false;
     }
 
+    template <int dim, typename Cell>
+    inline
+    double calculate_cell_radius(const Cell &cell)
+    {
+      double res = 0.;
+      for (unsigned int v=0; v<GeometryInfo<dim>::vertices_per_cell; ++v)
+        res = std::max(res, ( cell->vertex(v)).distance(cell->center()) );
```

Comment:
@davydden -- I don't think that the center does not have to lie on the largest diagonal. Are you sure?